app/bundles/CampaignBundle/Model/EventModel.php

```diff
@@ -1905,16 +1908,18 @@ public function invokeEventCallback($event, $settings, $lead = null, $eventDetai
      */
     public function checkEventTiming($action, \DateTime $parentTriggeredDate = null, $allowNegative = false)
     {
-        $now = new \DateTime();
-
-        $this->logger->debug('CAMPAIGN: Check timing for '.ucfirst($action['eventType']).' ID# '.$action['id']);
-
         if ($action instanceof Event) {
             $action = $action->convertToArray();
         }
+        $campaign = (array)$action['campaign'];
+        $campaignId = $campaign['id'];
```

Comment:
Using (array) will only return public properties from the object though right? so if $action['campaign'] is a Campaign entity, $campaign is going to be an empty array.